typings/connection.d.ts

```diff
@@ -0,0 +1,544 @@
+ * @interface AmqpMessage
+ */
+export interface AmqpMessage extends AmqpMessageProperties {
+  // TODO: Ask Gordon about other AMQP message properties like durable, first_acquirer, etc.
+  // https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-protocol-guide#messages
+  body: any;
+  message_annotations?: AmqpMessageAnnotations;
+  application_properties?: Dictionary<any>;
+  delivery_annotations?: DeliveryAnnotations;
+}
+
+/**
+ * Defines the AMQP Connection context. This context is provided when you add an
+ * event handler to any of the objects created by rhea.
+ * @interface Context
+ */
+export interface Context {
+  /**
+   * @property {Connection} connection The amqp connection.
+   */
+  connection: Connection;
+  /**
+   * @property {Container} container The amqp container
+   */
+  container: Container;
+  /**
+   * @property {Delivery} [delivery] The amqp delivery that is received after sending a message.
+   */
+  delivery?: Delivery;
+  /**
+   * @property {AmqpMessage} [message] The amqp message that is received in the message event
+   * handler when rhea emits a message event on a receiver.
+   */
+  message?: AmqpMessage;
+  /**
+   * @property {Receiver} [receiver] The amqp receiver link that was created on the amqp connection.
+   */
+  receiver?: Receiver;
+  /**
+   * @property {Session} session The amqp session link that was created on the amqp connection.
+   */
+  session: Session;
+  /**
+   * @property {Sender} [sender] The amqp sender link that was created on the amqp connection.
+   */
+  sender?: Sender;
+}
+
+/**
+ * Defines the amqp error object.
+ * @interface AmqpError
+ */
+export interface AmqpError {
+  /**
+   * @property {string} [condition] Describes the error condition.
+   */
+  condition?: string;
+  /**
+   * @property {string} [description] Describes any supplementary information that is not indicated the error condition.
+   */
+  description?: string;
+  /**
+   * @property {any} [info] Describes the information about the error condition.
+   */
+  info?: any;
+  /**
+   * @property {any[]} [value] Describes the associated amqp value types.
+   */
+  value?: any[];
+}
+
+export declare interface Connection extends EventEmitter {
+  [x: string]: any;
+  options: any;
+  hostname?: string;
+  container_id: string;
+  max_frame_size?: number;
+  idle_time_out?: number;
+  channel_max?: number;
+  registered: boolean;
+  state: EndpointState;
+  local_channel_map: { [x: string]: Session };
+  remote_channel_map: { [x: string]: Session };
+  local: any;
+  remote: any;
+  session_policy: any;
+  amqp_transport: Transport;
+  sasl_transport?: any;
+  conn_established_counter: number;
+  heartbeat_out: NodeJS.Timer;
+  heartbeat_in: NodeJS.Timer;
+  abort_idle: boolean;
+  socket_ready: boolean;
+  scheduled_reconnect?: NodeJS.Timer;
+  default_sender?: Sender;
+  properties?: { [x: string]: any };
+  readonly error?: any;
```

Comment:
What should be the type of the error property? AmqpError ? or a union of AmqpError and Error ?